src/FNAPlatform/SDL2_FNAPlatform.cs

```diff
@@ -844,24 +844,27 @@ public static void RunLoop(Game game)
 					// Text Input
 					else if (evt.type == SDL.SDL_EventType.SDL_TEXTINPUT && !textInputSuppress)
 					{
-						string text;
-
 						// Based on the SDL2# LPUtf8StrMarshaler
 						unsafe
 						{
 							byte* endPtr = evt.text.text;
-							while (*endPtr != 0)
+							if (*endPtr != 0)
 							{
-								endPtr++;
+								int bytes = 0;
+								while (*endPtr != 0)
+								{
+									endPtr++;
+									bytes++;
+								}
+								// utf8 will never encode more characters than bytes in a string so bytes is a suitable upper estimate of buffersize needed
+								char* charsBuffer = stackalloc char[bytes];
```

Comment:
Not NET40 compatible, maybe guard this with NETSTANDARD2_0 like we do with SDL2#:

https://github.com/flibitijibibo/SDL2-CS/blob/master/src/SDL2.cs#L84

Reply:
C# spec is written poorly and implies this requires modern C# when it doesn't, so it turns out this might be okay, I think. May have to test with BRUTE and Mono first.